File: Java-Core/src/main/java/presto/javascript/JavaScriptIntegerLiteral.java
package presto.javascript;

public class JavaScriptIntegerLiteral extends JavaScriptLiteral {

	Integer value;
	
	public JavaScriptIntegerLiteral(String text) {
		super(text);
		this.value = Integer.valueOf(text);
	}

	@Override
	public String toString() {
		return value.toString();
	}
}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptIntegerLiteral.js
var JavaScriptLiteral = require("./JavaScriptLiteral").JavaScriptLiteral;

function JavaScriptIntegerLiteral(text) {
	JavaScriptLiteral.call(this, text);
	return this;
}

JavaScriptIntegerLiteral.prototype = Object.create(JavaScriptLiteral.prototype);
JavaScriptIntegerLiteral.prototype.constructor = JavaScriptIntegerLiteral;

JavaScriptIntegerLiteral.prototype.check = function(context) {
	return IntegerType.instance;
};

JavaScriptIntegerLiteral.prototype.toString = function() {
	return this.value.toString();
};

exports.JavaScriptIntegerLiteral = JavaScriptIntegerLiteral;

